build/Helix/OutputTestResults.ps1

```diff
@@ -73,6 +74,14 @@ foreach ($testRun in ($testRuns.value | Sort-Object -Property "completedDate"))
                 $unreliableTests.Add($shortTestCaseTitle)
             }
         }
+        elseif ($testResult.outcome -ne "Passed")
```

Comment:
This is mostly a fix to error out in the case where something unexpected happens. But in this case what was happening was that we were leaving tests in the "NotExecuted" state, since we were skipping over them in UpdateUnreliableTests due to the duplicate run check. 

For example in this run:
https://dev.azure.com/ms/microsoft-ui-xaml/_build/results?buildId=33768&view=ms.vss-test-web.build-test-results-tab&runId=1680548&resultId=100001&paneView=debug
You can see that this test:
ReleaseTest.NugetTestsCX.RepeaterNoCrashTest
Is listed as "NotExecuted", which should not happen.